### PR TITLE
[feature] Add Twitter redirect endpoint

### PR DESCRIPTION
**Pull request by HarryRybacki**
## Purpose:
Allow individuals to get automatically redirected to a user's Twitter account if one is associated with the respective user.

## Changes:
- Add a new endpoint `/@<uid>/` that will automatically redirect to the Twitter account associated with `uid` if the account **is active** and **has a Twitter handle** associated with it. 
- If either of these conditionals is not met, the user is prompted with a concise error message.
screen shot 2015-07-01 at 2 01 46 pm
screen shot 2015-07-01 at 2 02 08 pm

## Side Effects:
A new route was added with a respective view.

### CODE REVIEW

File: website/profile/views.py [hunk 1]
```diff
@@ -784,3 +784,33 @@ def request_deactivation(auth):
...
+
+    :param uid: uid for requested User
+    :return: Redirect to User's Twitter account page
+    """
+    user = User.load(uid)
+
+    if user:
+        twitter_handle = user.social.get('twitter', None)
+    else:
+        raise HTTPError(http.BAD_REQUEST, data={
```
Comment (sloria) [inline website/profile/views.py]: A `NOT_FOUND` error is more appropriate here.

Comment (HarryRybacki) [inline website/profile/views.py]: Will modify, thanks!

**Comment by sloria** [comment]
After that minor change, this is good  to merge.

File: tests/test_views.py [hunk 2]
```diff
@@ -1414,6 +1414,36 @@ def test_unsubscribe_mailchimp_not_called_if_user_not_subscribed(self, mock_get_
...
+    def test_twitter_redirect_success(self):
+        self.user.social['twitter'] = 'someTwitterHandle'
+        self.user.save()
+        expected_redirect = 'https://twitter.com/someTwitterHandle'
+
+        res = self.app.get(web_url_for('redirect_to_twitter', uid=self.user._id))
+        assert_equals(res.status_code, 302)
+        assert_equal(res.location, expected_redirect)
+
+    def test_twitter_redirect_to_non_existent_user_returns_HTTPBad_request(self):
```
Comment (sloria) [inline tests/test_views.py]: This name is technically incorrect: a NOT FOUND *response* is returned. 

I'll make this change when I merge.

**Pull request merged**